### PR TITLE
[opichuryn] sequence_task_3

### PR DESCRIPTION
**Pull request by opichuryn**

### CODE REVIEW

File: sequence_task_3/student/include/impl.h [hunk 1]
```diff
@@ -7,46 +7,102 @@ template<class Container, class Song_t>
...
 
-    /** @todo Set track to be played next from lvalue */
-    void playNext(const Song_t& song);
+    /** Set track to be played next from lvalue */
+    void playNext(const Song_t& song) {
+        if (!hasTracks()) {
+            play(song);
+            return;
+        }
+        m_tracklist.push_front(song);
```
Comment (smay1613) [inline sequence_task_3/student/include/impl.h:70]: please check the difference with line 61 (insert position)

File: sequence_task_3/student/include/impl.h [hunk 2]
```diff
@@ -7,46 +7,102 @@ template<class Container, class Song_t>
...
+    using const_reference = typename Container::const_reference;
+    using iterator = typename Container::iterator;
+    using const_iterator = typename Container::const_iterator;
+    using difference_type = typename Container::difference_type;
+    using size_type = typename Container::size_type;
+    using self_type = DynamicPlaylist<Container, Song_t>;
 
-    /** @todo Iterators */
+    /** Iterators */
+    const_iterator begin() {
```
Comment (smay1613) [inline sequence_task_3/student/include/impl.h:21]: const

File: sequence_task_3/student/include/impl.h (hunk 1 above)
Comment (opichuryn) [reply sequence_task_3/student/include/impl.h]: Done!

File: sequence_task_3/student/include/impl.h (hunk 2 above)
Comment (opichuryn) [reply sequence_task_3/student/include/impl.h]: Done!

**Pull request closed without merging**